### PR TITLE
feat: Support auto-scaling per shard for cluster tier in `advanced_cluster`

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-289710

Merge auto-scaling per shard development branch into master.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by jwilliams-mongo — APPROVED** [review]

**Review by oarbusi — APPROVED** [review]

File: internal/service/advancedcluster/resource_advanced_cluster.go [hunk 1]
```diff
@@ -505,10 +505,26 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
...
 	}
-	// cannot call latest API (2024-10-23 or newer) as it can enable ISS autoscaling
-	cluster, _, err := connV220240805.ClustersApi.CreateCluster(ctx, projectID, params).Execute()
-	if err != nil {
-		return diag.FromErr(fmt.Errorf(errorCreate, err))
+
+	var clusterName string
+	var clusterID string
+	var err error
+	if isUsingOldShardingConfiguration(d) {
```
Comment (EspenAlbert) [inline internal/service/advancedcluster/resource_advanced_cluster.go:513]: Include old comment? // cannot call latest API (2024-10-23 or newer) as it can enable ISS autoscaling

File: internal/service/advancedcluster/resource_advanced_cluster.go [hunk 2]
```diff
@@ -584,72 +600,39 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
...
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
 		}
+		return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
+	}
 
-		zoneNameToZoneIDs, err := getZoneIDsFromNewAPI(cluster)
-		if err != nil {
+	zoneNameToOldReplicationSpecMeta, err := GetReplicationSpecAttributesFromOldAPI(ctx, projectID, clusterName, connV220240530.ClustersApi)
```
Comment (EspenAlbert) [inline internal/service/advancedcluster/resource_advanced_cluster.go:615]: [nit] Rename to oldShardConfigMeta?

File: internal/service/advancedcluster/resource_advanced_cluster.go [hunk 3]
```diff
@@ -1462,5 +1430,6 @@ func waitForUpdateToFinish(ctx context.Context, connV2 *admin.APIClient, project
 	}
 
 	_, err := stateConf.WaitForStateContext(ctx)
+
```
Comment (EspenAlbert) [inline internal/service/advancedcluster/resource_advanced_cluster.go]: [nit] extra \n

File: internal/service/advancedclustertpf/schema.go [hunk 4]
```diff
@@ -544,8 +544,8 @@ type TFModel struct {
 	Labels                                    types.Set      `tfsdk:"labels"`
 	ReplicationSpecs                          types.List     `tfsdk:"replication_specs"`
 	Tags                                      types.Set      `tfsdk:"tags"`
-	StateName                                 types.String   `tfsdk:"state_name"`
-	ConnectionStrings                         types.Object   `tfsdk:"connection_strings"`
+	MongoDBVersion                            types.String   `tfsdk:"mongo_db_version"`
```
Comment (EspenAlbert) [inline internal/service/advancedclustertpf/schema.go]: Why these changes?

**Review by EspenAlbert — APPROVED** [review]
LGTM!

File: internal/service/advancedcluster/resource_advanced_cluster.go (hunk 3 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster.go]: removed

File: internal/service/advancedclustertpf/schema.go (hunk 4 above)
Comment (AgustinBettati) [reply internal/service/advancedclustertpf/schema.go]: not certain how they were caused, added a commit to align ordering of fields as defined in master 👍

File: internal/service/advancedcluster/resource_advanced_cluster.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster.go:513]: good point, added clarifying comments

File: internal/service/advancedcluster/resource_advanced_cluster.go (hunk 2 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster.go:615]: while more verbose I do like the current name as it give insight into how the resulting map is composed (zoneName -> replication spec metadata)
